be/src/Project/Services/Catalog/Catalog.Api/Services/Products/ProductsService.cs

```diff
@@ -99,9 +100,13 @@ public class ProductsService : IProductsService
                     ProductId = product.Id
                 };
 
-                await this.catalogContext.ProductImages.AddAsync(productImage.FillCommonProperties());
+                images.Add(productImage.FillCommonProperties());
             }
 
+            await _context.ProductImages.AddRangeAsync(images);
```

Comment:
**suggestion:** the same as before, let's abandon redundant List<ProductImage>()